webextensions/common/common.js

```diff
@@ -334,6 +337,41 @@ configs.$loaded.then(() => {
 });
 
 
+export function filterMap(arr, callback, thisArg) {
+  if (arr == null) {
+    throw new TypeError('arr is null or undefined');
+  }
+
+  arr = Object(arr);
+  const maxi = arr.length >>> 0;
+
+  if (typeof callback !== 'function')
+    throw new TypeError(`${callback} is not a function`);
+
+  callback = callback.bind(thisArg == undefined ? arr : thisArg);
+
+  const newArr = new Array(maxi);
+  let counti = 0,
+      i = 0,
+      value,
+      newValue;
+
+  while (i < maxi) {
+    if (i in arr) {
+      value = arr[i];
+      newValue = callback(value, i, arr);
+      if (newValue !== undefined) {
```

Comment:
I'm afraid that I may forget to return `undefined` by the callback function, because I ordinarily ignore difference of `null`, `false` and `undefined`. Are `!newValue` or `newValue !== undegined && newValue !== false && newValue !== null` painfully expensive than `!== undefined`?